src/compat.h

```diff
@@ -96,6 +96,12 @@ typedef int32_t ssize_t;
 size_t strnlen( const char *start, size_t max_len);
 #endif // HAVE_DECL_STRNLEN
 
+#ifndef WIN32
+typedef void* sockopt_arg_t;
```

Comment:
Type names ending in `_t` are technically reserved by POSIX, so risk colliding with system types on some architectures.